src/main/java/org/openlmis/requisition/web/BaseRequisitionController.java

```diff
@@ -321,6 +321,11 @@ ETagResource<RequisitionDto> doUpdate(Requisition requisitionToUpdate, Requisiti
     Map<VersionIdentityDto, ApprovedProductDto> approvedProducts = findApprovedProducts(
         requisitionToUpdate::getAllApprovedProductIdentities, profiler);
 
+    if (Boolean.FALSE.equals(requisitionToUpdate.getEmergency())) {
```

Comment:
Why it's not somewhere in org.openlmis.requisition.web.BaseRequisitionController.UpdateParams#updateAndSave ?

Or even deeper in: org.openlmis.requisition.domain.requisition.Requisition#updateFrom

Reply:
The problem appears earlier, in BatchRequisitionController, [line 274](https://github.com/OpenLMIS/openlmis-requisition/blob/43ba11354c88ed0e231290b2820d833014ed8674/src/main/java/org/openlmis/requisition/web/BatchRequisitionController.java#L274), where the requisition is built, based on which requisitionToUpdate is later updated. Already in the process of this build, the processingPeriodId and facilityId are 'lost'. Therefore, we could also put these setters in the buildRequisition method of BatchRequisitionController or RequisitionBuilder.newRequisition (I didn't want to put it here due to the use of this method in other places and possible complications). Do you think we should change this and put the setters e.g. in the buildRequisition method?

Reply:
I see, then I would move it to `org.openlmis.requisition.domain.requisition.RequisitionBuilder#newRequisition`, mind the `processingPeriodDto` argument of `newRequisition`. 
OR I you think it too risky in OLMIS, then don't change it and just merge the PR.

Reply:
considering other uses of this method and the need to modify it I will leave it as is